File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <stack>
using namespace std;


stack<int>s1;

int main(){



    int N ;
    cin >> N;


    for(int i = 0 ; i < N ; i ++){
        string s;
        cin >> s;
        if(s=="push"){
            int a=  0;
            cin >> a;
            s1.push(a);

        }
        if(s=="pop"){

            if(s1.size()>0){
                cout << s1.top() << "\n";
                s1.pop();

            }else{
                cout << -1<<"\n";
            }
        }
        if(s=="size"){
            cout << s1.size()<<"\n";
        }
        if(s=="empty"){
            if(s1.size()==0){
                cout << 1<<"\n";
            }else{
                cout << 0 << "\n";
            }

        }
        if(s=="top"){
            if(s1.size()>0){
                cout << s1.top() << "\n";
            }else{
                cout << -1 << "\n";
            }

        }


    }

}

File: 백준/Silver/10828. 스택/스택.java
import java.io.*;
import java.sql.Array;
import java.util.*;


public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
       // BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out)); // 선언

        Stack<Integer> stack = new Stack<>();


        int n = Integer.parseInt(br.readLine());

        for(int i = 0 ; i < n ; i ++ ){

            String s  = br.readLine();

            String ss = s.split(" ")[0];

            if(ss.equals("push")){
                stack.push(Integer.parseInt(s.split(" ")[1]) );
            }
            int sn=0;

            switch(ss){

                case "pop":
                    try{
                        sn = stack.pop();
                        System.out.println(sn);
                    }catch(Exception e){
                        System.out.println(-1);
                    }
                    break;

                case "size":
                    System.out.println(stack.size());
                    break;

                case "empty":
                    if(stack.isEmpty()){
                        System.out.println(1);
                    }else{
                        System.out.println(0);
                    }
                    break;
                case "top":
                    try{
                        sn = stack.peek();
                        System.out.println(sn);

                    }catch(Exception e ){
                        System.out.println(-1);
                    }
                    break;




            }



        }











        }
    }


